tests/projects/test_virtualenv_projects.py

```diff
@@ -33,17 +33,18 @@ def use_dev_mlflow_for_projects():
         conda_pip_dependencies.append(mlflow_root)
     write_yaml(TEST_VIRTUALENV_CONDA_PROJECT_DIR, "conda.yaml", conda_env, overwrite=True)
 
-    virtualenv_requirements_path = os.path.join(TEST_VIRTUALENV_PROJECT_DIR, "requirements.txt")
-    with open(virtualenv_requirements_path) as f:
-        virtualenv_requirements = f.readlines()
-
-    with open(virtualenv_requirements_path, "w") as f:
-        for line in virtualenv_requirements:
-            if line.rstrip("\n") != "mlflow":
-                f.write(line)
-            else:
-                f.write(mlflow_root)
-                f.write("\n")
+    for proj_dir in (TEST_VIRTUALENV_PROJECT_DIR, TEST_VIRTUALENV_NO_PYTHON_ENV):
```

Comment:
`test_virtualenv_project_execution_no_python_env` uses `TEST_VIRTUALENV_NO_PYTHON_ENV`. `requirements.txt` in this directory needs to be modified to install mlflow from the local repository, not PyPI.